src/NServiceBus.RabbitMQ/Configuration/ConnectionStringParser.cs

```diff
@@ -46,6 +46,19 @@ public ConnectionConfiguration Parse(string connectionString)
                 throw new Exception("Invalid connection string. 'host' value must be supplied. e.g: \"host=myServer\"");
             }
 
+            if (settings.HasSetting(SettingsKeys.UsePublisherConfirms))
```

Comment:
While the `ConnectionStringParser` does have access to the settings (really only to pass it to `ConnectionConfiguration`, which in turn only needs it for the endpoint name) doing this check here feels wrong to me. Overall, it feels like RabbitMQTransportInfrastructure would be a better place for this...